text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1061 @@
+
+The host pre-built standard library will always be used for procedural macros
+and build scripts ([?][rationale-host-deps-cross],
+[?][rationale-host-deps-host]). Multi-target projects (resulting from the
+`target` field in artifact dependencies or the use of `per-pkg-target` fields)
+may result in the standard library being built multiple times - once for each
+target in the project.
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why put `build-std` in the Cargo config?*][rationale-build-std-in-config]
+- [*Why accept `never` as a value for `build-std`?*][rationale-build-std-never]
+- [*Why add `build-std` to the `[target.<triple>]` and `[target.<cfg>]` sections?*][rationale-build-std-target-section]
+- [*Why does `[target]` take precedence over `[build]` for `build-std`?*][rationale-build-std-precedence]
+- [*Why does "always" rebuild unconditionally?*][rationale-unconditional]
+- [*Why does "always" rebuild in release profile?*][rationale-release-profile]
+- [*Why add `build-std-crate`?*][rationale-build-std-crate]
+- [*Why use the lockfile of the `rust-src` component?*][rationale-lockfile]
+- [*Why not build the standard library in incremental?*][rationale-incremental]
+- [*Why not produce a `dylib` for the standard library?*][rationale-no-dylib]
+- [*Why use the pre-built standard library for procedural macros and build scripts in host mode?*][rationale-host-deps-host]
+- [*Why use the pre-built standard library for procedural macros and build scripts in cross-compile mode?*][rationale-host-deps-cross]
+
+*See the following sections for relevant unresolved questions:*
+
+- [*What should the `build-std` configuration in `.cargo/config` be named?*][unresolved-config-name]
+- [*What should the "always" and "never" values of `build-std` be named?*][unresolved-config-values]
+- [*What should `build-std-crate` be named?*][unresolved-build-std-crate-name]
+- [*Should the standard library inherit RUSTFLAGS?*][unresolved-inherit-rustflags]
+
+## Interactions with `#![no_std]`
+[interactions-with-no_std]: #interactions-with-no_std
+
+Behaviour of crates using `#![no_std]` will not change whether or not `std` is
+rebuilt and passed via `--extern` to rustc, and `#![no_std]` will still be
+required in order for `rustc` to not attempt to load `std` and add it to the
+extern prelude.
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not replace `#![no_std]` as the source-of-truth for whether a crate depends on `std`?*][rationale-replace-no_std]
+
+## `restricted_std`
+[restricted_std]: #restricted_std
+
+The existing `restricted_std` mechanism will be removed from the standard
+library's [`build.rs`][std-build.rs].
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why remove `restricted_std`?*][rationale-remove-restricted-std]
+
+## Custom targets
+[custom-targets]: #custom-targets
+
+Cargo will detect when the standard library is to be built for a custom target
```

Comment:
Might be worth clarifying here that this only applies on stable channel releases of Rust.

Reply:
We do say that in the paragraph below:

> Custom targets can still be used with build-std on nightly toolchains provided
that `-Zunstable-options` is provided to Cargo.